### PR TITLE
Additon Of Slug Column

### PR DESCRIPTION
**Pull request by kunal202**
Fix Issue #78 
- [x]  Write a migration to add slug string column to calendar_events table which can be null.
- [x] Update the sequelize model accordingly
- [x] Update CalendarEvent graphql type
- [x] Update CalendarEventCreate mutation to auto generate slug from the event title using slugify.
- [x] Update CalendarEventUpdate mutation to support slug update.
- [x] Integrate the new column in the view and edit event modals. In the edit modal show an input field below description to edit the slug. The input value should be initialized by the slug of the event.

# TodO 
- [ ] Update the AuthorizationPolicy to allow update of the column to organizer and admins and allow read to everyone.
- [x] In the view modal, add the slug text at the top linking to /events/<slug>

### CODE REVIEW

**Comment by 0xnmn** [comment]
add package-lock.json to gitignore.

**Comment by kunal202** [comment]
I am unable to update the authorizaton Policy for slug , please guide

File: api/services/AuthorizationPolicy/index.js [hunk 1]
```diff
@@ -85,11 +86,17 @@ policy.include('calendarEvent', (p) => {
...
+        if (viewer.role === 'Admin') return true;
+
+        if (viewer.id === event.id) {
+          if (value.role === 'Admin') return false;
+
+          return true;
+        }
+
+        return false;
+      });
```
Comment (0xnmn) [inline api/services/AuthorizationPolicy/index.js]: This is not required.

**Comment by 0xnmn** [comment]
> I am unable to update the authorizaton Policy for slug , please guide

You just need to add https://github.com/coding-blocks/gondor/pull/85/files#diff-4346abda165476a1c2395737eadfb3f4R79, which you have already done.

For more info, read docs from here: https://github.com/hereisnaman/auth-policy

File: api/services/AuthorizationPolicy/index.js (hunk 1 above)
Comment (kunal202) [reply api/services/AuthorizationPolicy/index.js]: Okay

**Comment by 0xnmn** [comment]
@kunal202 the slugify logic should be at the API side. No need to accept slug in the create event mutation. Auto generate the slug and add it while creating an event here: https://github.com/coding-blocks/gondor/blob/master/api/services/CalendarEvent/index.js#L11

**Comment by kunal202** [comment]
I am getting 400 response while submitting the form, any reason for it 

video

**Comment by 0xnmn** [comment]
Issue closed.

**Pull request closed without merging**